x/warp/config.go

```diff
@@ -0,0 +1,213 @@
+	return &Config{
+		Upgrade:         precompileconfig.Upgrade{BlockTimestamp: blockTimestamp},
+		QuorumNumerator: quorumNumerator,
+	}
+}
+
+// NewDefaultConfig returns a config for a network upgrade at [blockTimestamp] that enables
+// Warp with the default quorum numerator (0 denotes using the default).
+func NewDefaultConfig(blockTimestamp *big.Int) *Config {
+	return NewConfig(blockTimestamp, 0)
+}
+
+// NewDisableConfig returns config for a network upgrade at [blockTimestamp]
+// that disables Warp.
+func NewDisableConfig(blockTimestamp *big.Int) *Config {
+	return &Config{
+		Upgrade: precompileconfig.Upgrade{
+			BlockTimestamp: blockTimestamp,
+			Disable:        true,
+		},
+	}
+}
+
+// Key returns the key for the Warp precompileconfig.
+// This should be the same key as used in the precompile module.
+func (*Config) Key() string { return ConfigKey }
+
+// Verify tries to verify Config and returns an error accordingly.
+func (c *Config) Verify() error {
+	if c.QuorumNumerator > params.WarpQuorumDenominator {
+		return fmt.Errorf("cannot specify quorum numerator (%d) > quorum denominator (%d)", c.QuorumNumerator, params.WarpQuorumDenominator)
+	}
+	// If a non-default quorum numerator is specified and it is less than the minimum, return an error
+	if c.QuorumNumerator != 0 && c.QuorumNumerator < params.WarpQuorumNumeratorMinimum {
+		return fmt.Errorf("cannot specify quorum numerator (%d) < min quorum numerator (%d)", c.QuorumNumerator, params.WarpQuorumNumeratorMinimum)
+	}
+	return nil
+}
+
+// Equal returns true if [s] is a [*Config] and it has been configured identical to [c].
+func (c *Config) Equal(s precompileconfig.Config) bool {
+	// typecast before comparison
+	other, ok := (s).(*Config)
+	if !ok {
+		return false
+	}
+	equals := c.Upgrade.Equal(&other.Upgrade)
+	return equals && c.QuorumNumerator == other.QuorumNumerator
+}
+
+func (c *Config) Accept(acceptCtx *precompileconfig.AcceptContext, txHash common.Hash, logIndex int, topics []common.Hash, logData []byte) error {
+	unsignedMessage, err := warp.ParseUnsignedMessage(logData)
+	if err != nil {
+		return fmt.Errorf("failed to parse warp log data into unsigned message (TxHash: %s, LogIndex: %d): %w", txHash, logIndex, err)
+	}
+	log.Info("Accepted warp unsigned message", "txHash", txHash, "logIndex", logIndex, "logData", common.Bytes2Hex(logData))
+	if err := acceptCtx.Warp.AddMessage(unsignedMessage); err != nil {
+		return fmt.Errorf("failed to add warp message during accept (TxHash: %s, LogIndex: %d): %w", txHash, logIndex, err)
+	}
+	return nil
+}
+
+// verifyWarpMessage checks that [warpMsg] can be parsed as an addressed payload and verifies the Warp Message Signature
+// within [predicateContext].
+func (c *Config) verifyWarpMessage(predicateContext *precompileconfig.ProposerPredicateContext, warpMsg *warp.Message) error {
+	// Use default quorum numerator unless config specifies a non-default option
+	quorumNumerator := params.WarpDefaultQuorumNumerator
```

Comment:
instead of this, can we configure the default in module.go/#Configure and set to `c.QuorumNumerator`?

Reply:
related to https://github.com/ava-labs/subnet-evm/pull/718#discussion_r1271963249

Reply:
I think it's simpler to keep this entirely in-memory on the config for warp itself and provide that from the config during verification, so that we don't need to write it into the state. I think writing it in the state makes sense when we want it to be mutable via on-chain events.